### PR TITLE
tuple to struct event variants

### PR DESCRIPTION
**Pull request by Doordashcon**
This pull request changes the pallet Events variants from tuples to structs with named fields.

Part implementation for #9903, #9993 & #10184

## Pallets with changes
- sudo
- tips

polkadot address: 12zsKEDVcHpKEWb99iFt3xrTCQQXZMu477nJQsTBBrof5k2h

### CODE REVIEW

File: utils/frame/remote-externalities/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jsonrpsee-ws-client = { version = "0.3.1", default-features = false, features =
 jsonrpsee-proc-macros = "0.3.1"
 
 env_logger = "0.9"
+frame-support = { path = "../../../frame/support" }
```
Comment (shawntabrizi) [inline utils/frame/remote-externalities/Cargo.toml:22]: whats this?

**Review by shawntabrizi — APPROVED** [review]

File: utils/frame/remote-externalities/Cargo.toml (hunk 1 above)
Comment (Doordashcon) [reply utils/frame/remote-externalities/Cargo.toml:22]: while running "cargo +nightly test --all-feature"  I got  "failed to resolve: use of undeclared crate or module `frame_support`
   --> utils/frame/remote-externalities/src/lib.rs:644:3"
so I added "frame-suppport" by path to the Cargo.toml

**Comment by Doordashcon** [comment]
{polkadot} address: {12zsKEDVcHpKEWb99iFt3xrTCQQXZMu477nJQsTBBrof5k2h}

**Comment by KiChjang** [comment]
@Doordashcon you need to put it in your PR description, not as a comment. I've edited your PR post to include your DOT address. Also take note of the format, i.e. no braces are required.

File: utils/frame/remote-externalities/Cargo.toml (hunk 1 above)
Comment (KiChjang) [reply utils/frame/remote-externalities/Cargo.toml:22]: This should only be required when the `remote-test` feature is enabled, so I would put this as an optional dependency and only enable it when the `remote-test` feature is enabled.

**Review by KiChjang — APPROVED** [review]

**Comment by Doordashcon** [comment]
> /tip small

Thanks for the tip, I have updated my Polkadot Address

**Review by shawntabrizi — APPROVED** [review]

**Comment by KiChjang** [comment]
Need to run `cargo +nightly fmt`.

**Comment by KiChjang** [comment]
bot merge

**Comment by xlc** [comment]
Do we know if this is going to cause any compatibility concerns on client side?

**Pull request merged**